### PR TITLE
First shitty draft

### PR DESCRIPTION
**Pull request by xkasianx**
First shitty draft of the design doc. You can ignore my notes on the article

### CODE REVIEW

File: module-1/desgindoc.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# Motivation
+Currently, users can explore new goods from friends feed, chronological feed or one for all users curated weekly feed.
+
+**Hypothesis:** If we implement For You feed with personalized recommendations we will improve user experience in exploring new items, increase engagement, retention rate and session time.
+
+**Success metrics:**
+Increased engagement (DAU, session time, retention rate)
+
+# Requirements & Constraints
+Should work only on users with sufficient data(>10 goods, tbd). Cold-start problem is out of scope.
```
Comment (truskovskiyk) [inline module-1/desgindoc.md:24]: why 10? is it a guess or is something else behind it?

File: module-1/desgindoc.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+- Users should receive a personalized feed with at least 5 recommended items per day, based on their past activity, saved products, and interaction with other users' collections.
+- **Cost Efficiency:** not crazy cost's in terms of compute(<$25-50/month), models desirably should run on CPU
+- **Scalability:** The recommendation system should be designed to scale as the user base grows, ensuring that the system remains performant even as new users and more complex datasets are added.
+
+Current goods count: ~100,000
+Current users count: ~5000
```
Comment (truskovskiyk) [inline module-1/desgindoc.md:31]: where are those coming from?

File: module-1/desgindoc.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+# Data
+Chronological user goods sequence(Good#1 -> Good#3 -> Good#345)
+
+Goods(image, MobileClip image embedding, title, url(can extract domain, or instagram handle))
+
+# Techniques
+## Tried(prototyped fast, might need to try again):
+- finding closest good to an average of all user goods Clip embeddings. performed poorly, lacked diversity
+- clustering user's goods embeddings, and recomending closest to each cluster average. also preformed poorly(might explore why)
+- doing svd on collaborative matrix. performed poorly
```
Comment (truskovskiyk) [inline module-1/desgindoc.md:45]: `performed poorly`? what was the metric?

File: module-1/desgindoc.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+I see potential in using chronological goos sequence as predicting next good. (Good#1 -> Good#3 -> Good#345)
+- predicting next item in goods embedding sequence using transformer/rnn
+- ? graph nn's, pinterest uses them
+- explore papers for user-to-item prediction
+
+# Evalutaion
+- Do a time-based split on user's goods and evaluate how well model predicts next goods(or a vector that is similar to the next goods).
+- Visual vibe check, how well product team likes recommendations
+
+ChatGPT said: Precision @ K, Recall @ K, Diversity Score. Need to explore where to get data or label data for the evaluation.
```
Comment (truskovskiyk) [inline module-1/desgindoc.md:57]: looks like a retrieval problem to me: https://en.wikipedia.org/wiki/Evaluation_measures_(information_retrieval)

File: module-1/desgindoc.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+TBD: Decide if system will need regular model retraining/fine-tuning or one time.
+
+## Data-Flow Diagram
+User actions(saving/collecting good) ->
+Data Storage(PostgreSQL) ->
+Cron Job triggered ->
+Preprocessing ->
+Rec System(Model training/inference, candidate searching, ranking) ->
+Save recs to db ->
+For You Page
```
Comment (truskovskiyk) [inline module-1/desgindoc.md:69]: might be easier if this was in google doc

**Review by truskovskiyk — APPROVED** [review]

File: module-1/desgindoc.md (hunk 1 above)
Comment (xkasianx) [reply module-1/desgindoc.md:24]: more of a guess

File: module-1/desgindoc.md (hunk 2 above)
Comment (xkasianx) [reply module-1/desgindoc.md:31]: i run a query to count the users and goods in db :)

File: module-1/desgindoc.md (hunk 3 above)
Comment (xkasianx) [reply module-1/desgindoc.md:45]: vibe check :)))

**Pull request merged**